include/crm/common/ordering.h

```diff
@@ -31,6 +31,12 @@ enum pe_ordering {
     //! Ordering applies only if first action cannot be related to a migration
     pcmk_ord_if_first_unmigratable      = (1U << 1),
 
+    /*!
+     * If 'then' is required, 'first' becomes required; if 'first' is a stop of
+     * a blocked resource, 'then' becomes unrunnable
```

Comment:
Another thing that's confusing is that internally, ordering flags are used for any relationship between actions, not just ordering (first_else_then being a good example). I wonder whether we should use `pcmk_rel_` instead of `pcmk_ord_`.

Technically the "blocked stop" part applies regardless of whether either action is required, though in practice I think it could only matter if they are.

Worth adding "If then is unmigratable, first becomes unmigratable"

Reply:
`pcmk_rel_` would make sense. The one point against `pcmk_rel` is that it's less clear that the relationships are specific to actions. An ordering could only reasonably apply to actions, while a relationship could apply to any two objects. I don't think that's a big deal.

Reply:
Yeah, not ideal. Collective resources are a relationship between resources, a location constraint is a relationship between a resource and a node, and so on. But if we only use relationship as a formal term in respect to actions, it might be acceptable. "Action relationship" is better but `pcmk_actrel_` is not particularly readable and already long for a prefix. `pcmk_ar_` is barely readable but a great prefix. Maybe `ord` stands for "optimal relationship directive" ;)

Reply:
I think I'm just going to make all of this internal to libcrmcommon and keep the old names strictly as "do not use" deprecated synonyms. I don't think any external code needs this level of detail into the ordering code. It's going to be painful to redo :/ but there's no point in doing all of this just to pull it internal with 3.0.0. The naming becomes less important in that case, though still relevant for our own benefit. I'm thinking `pcmk__ar_`.